spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -327,6 +328,7 @@ private Result doExecuteWithPartialProgress(RewriteExecutionContext ctx, Stream<
       LOG.error("{} is true but no rewrite commits succeeded. Check the logs to determine why the individual " +
           "commits failed. If this is persistent it may help to increase {} which will break the rewrite operation " +
           "into smaller commits.", PARTIAL_PROGRESS_ENABLED, PARTIAL_PROGRESS_MAX_COMMITS);
+      throw new NoCommitSucceedException("No rewrite commits succeeded after max commits");
```

Comment:
Why not throw `CommitFailedException`? That should be the underlying exception that caused all of the failures, right?

Reply:
I mentioned this above, but we won't get CommitFailedExceptions for a bunch of other cases. Like say the user lacks permission to write to storage, or there is a network error while a shuffle is occurring.

Reply:
Sounds good. Commit failed is not a good idea then. What else could we throw instead of creating a new exception?

Reply:
Yeah i'm not sure, I feel like this is a common pain point in asynchronous failures. Ideally I'd like to gather all unique error messages and rethrow them, but I've previously dealt with this by just keeping a "last thrown" exception and just rethrowing that.

Reply:
@rdblue I see that we have plans to rewrite delete files, if we support partial commit for rewrite delete files, it also will have a no-commit-succeed scenario. So i think maybe it makes sense to create a new exception.